File: t/CS/MedicalViewerDemo/UI/TextAnnotationDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools;
using Leadtools.MedicalViewer;

namespace MedicalViewerDemo
{
   public partial class TextAnnotationDialog : Form
   {
      MedicalViewerAnnotationText _textAnnotation;
      MedicalViewerCell _cell;
      MedicalViewer _viewer;

      public TextAnnotationDialog(MainForm owner)
      {
         InitializeComponent();

         _cell = MainForm.DefaultCell;
         _viewer = owner.Viewer;

         if (_viewer.Cells.Count ==  0)
            _textAnnotation = (MedicalViewerAnnotationText)(_cell.GetActionProperties(MedicalViewerActionType.AnnotationText));
         else
            _textAnnotation = (MedicalViewerAnnotationText)(_viewer.Cells[0].GetActionProperties(MedicalViewerActionType.AnnotationText));

         _lblColor.BackColor = Color.FromArgb(0xff, _textAnnotation.AnnotationColor);
         _textAnnotation.TextColor = Color.FromArgb(0xff, _textAnnotation.TextColor);
         _cmbApplyTo.SelectedIndex = (int)_textAnnotation.Flags;
      }

      private void _btnOK_Click(object sender, EventArgs e)
      {
         this.Close();
         _btnApply_Click(sender, e);
      }

      private void _btnApply_Click(object sender, EventArgs e)
      {
         _textAnnotation.Flags = ((MedicalViewerAnnotationFlags)_cmbApplyTo.SelectedIndex);
         _textAnnotation.AnnotationColor = _lblColor.BackColor;
         _textAnnotation.TextColor = _lblColor.BackColor;

         _cell.SetActionProperties(MedicalViewerActionType.AnnotationText, _textAnnotation);

         foreach (MedicalViewerMultiCell viewerCell in _viewer.Cells)
         {
            viewerCell.SetActionProperties(MedicalViewerActionType.AnnotationText, _textAnnotation);
         }
      }

      private void _btnColor_Click(object sender, EventArgs e)
      {
         MainForm.ShowColorDialog(_lblColor);
      }
   }
}

File: t/VB/MedicalViewerDemo/UI/TextAnnotationDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Leadtools
Imports Leadtools.MedicalViewer

Namespace MedicalViewerDemo
   Partial Public Class TextAnnotationDialog : Inherits Form
      Private _textAnnotation As MedicalViewerAnnotationText
      Private _cell As MedicalViewerCell
      Private _viewer As MedicalViewer

      Public Sub New(ByVal owner As MainForm)
         InitializeComponent()

         _cell = MainForm.DefaultCell
         _viewer = owner.Viewer

         If _viewer.Cells.Count = 0 Then
            _textAnnotation = CType(_cell.GetActionProperties(MedicalViewerActionType.AnnotationText), MedicalViewerAnnotationText)
         Else
            _textAnnotation = CType(_viewer.Cells(0).GetActionProperties(MedicalViewerActionType.AnnotationText), MedicalViewerAnnotationText)
         End If

         _lblColor.BackColor = Color.FromArgb(&HFF, _textAnnotation.AnnotationColor)
         _textAnnotation.TextColor = Color.FromArgb(&HFF, _textAnnotation.TextColor)
         _cmbApplyTo.SelectedIndex = CInt(_textAnnotation.Flags)
      End Sub

      Private Sub _btnOK_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnOK.Click
         Me.Close()
         _btnApply_Click(sender, e)
      End Sub

      Private Sub _btnApply_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnApply.Click
         _textAnnotation.Flags = (CType(_cmbApplyTo.SelectedIndex, MedicalViewerAnnotationFlags))
         _textAnnotation.AnnotationColor = _lblColor.BackColor
         _textAnnotation.TextColor = _lblColor.BackColor

         _cell.SetActionProperties(MedicalViewerActionType.AnnotationText, _textAnnotation)

         For Each viewerCell As MedicalViewerMultiCell In _viewer.Cells
            viewerCell.SetActionProperties(MedicalViewerActionType.AnnotationText, _textAnnotation)
         Next viewerCell
      End Sub

      Private Sub _btnColor_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnColor.Click
         MainForm.ShowColorDialog(_lblColor)
      End Sub
   End Class
End Namespace
